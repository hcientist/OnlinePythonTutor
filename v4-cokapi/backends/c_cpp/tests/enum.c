// from a bug report:
enum color
{
  RED,
  ORANGE,
  YELLOW,
  GREEN,
  BLUE,
  INDIGO,
  VIOLET,
};

int main() {
  enum color my_color = RED;
  return 0;
}
