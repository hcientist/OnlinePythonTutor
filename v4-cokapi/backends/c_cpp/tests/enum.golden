{
  "code": "// from a bug report:\nenum color\n{\n  RED,\n  ORANGE,\n  YELLOW,\n  GREEN,\n  BLUE,\n  INDIGO,\n  VIOLET,\n};\n\nint main() {\n  enum color my_color = RED;\n  return 0;\n}", 
  "trace": [
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 14, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "my_color": [
              "C_DATA", 
              "0xFFF00010C", 
              "color", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 14, 
          "ordered_varnames": [
            "my_color"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 15, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "my_color": [
              "C_DATA", 
              "0xFFF00010C", 
              "color", 
              0
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 14, 
          "ordered_varnames": [
            "my_color"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 16, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "my_color": [
              "C_DATA", 
              "0xFFF00010C", 
              "color", 
              0
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 16, 
          "ordered_varnames": [
            "my_color"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }
  ]
}

